grace/base.py

```diff
@@ -37,27 +37,48 @@ class Annotation(enum.IntEnum):
 
 @dataclasses.dataclass
 class Prediction:
-    """Value of predicted label (argmax) of all
-    normalised class probabilities (softmax).
+    """Prediction dataclass all normalised softmax class probabilities.
 
     Parameters
     ----------
-    LABEL : int
-        Predicted class label (argmax)
-    PROB_TN : float
-        Probability of true negative detection (softmax)
-    PROB_TP : float
-        Probability of true positive detection (softmax)
+    softmax_probabs : npt.NDArray
+        Array or normalised softmax probs as predicted by classifier.
+
+    Methods
+    -------
+    label : self -> Annotation
+        Annotation class label with the highest probability.
+    prob_TN : self -> float
+        Probability of true negative detection (normalised softmax).
+    prob_TP : self -> float
+        Probability of true positive detection (normalised softmax).
 
     Notes
     -----
     - Normalised probabilities of all classes must sum up to 1.
-    - LABEL is the index of the highest softmax label probability.
+    - label return the Annotation (index) of the highest label prob.
     """
 
-    LABEL: int
-    PROB_TN: float
-    PROB_TP: float
+    softmax_probabs: npt.NDArray
+
+    def __post_init__(self):
+        assert self.softmax_probabs.ndim == 1
+        assert self.softmax_probabs.shape[0] >= 2
```

Comment:
```suggestion
        assert self.softmax_probabs.shape[0] == len(Annotation)
```

I would make this explicit, the array needs to be the same as the number of possible labels

Reply:
With the current implementation of the classifier, which outputs 2 labels and ignores `UNKNOWN`, perhaps it's better to check for `len(Annotation)` or `len(Annotation) - 1`...?